src/world.rs

```diff
@@ -0,0 +1,214 @@
+use std::{
+    fs, io::Read, path::{Path, PathBuf}
+};
+
+use regex::Regex;
+use serde::Deserialize;
+
+use crate::{Error, Map, ResourceCache, ResourceReader};
+
+/// A World is a collection of maps and their layout in the game world.
+#[derive(Deserialize, PartialEq, Clone, Debug)]
+pub struct World {
+    /// The path first used in a ['ResourceReader'] to load this world.
```

Comment:
```suggestion
    /// The path first used in a [`ResourceReader`] to load this world.
```
Backticks are important to make the link to the relevant item in documentation.